src/modules/kubernetes.rs

```diff
@@ -161,38 +198,97 @@ pub fn module<'a>(context: &'a Context) -> Option<Module<'a>> {
         None => false,
     });
 
+    // Parse config under `display`.
+    // Select the first style that matches the context_pattern and,
+    // if it is defined, the user_pattern
+    let matched_context_config = config.contexts.iter().find(|context_config| {
```

Comment:
I'm find the computation of this variable's value hard to read. I don't know if the fact that I'm a little tired right now affects it, but there's a lot of error handling, logging, short-circuiting, and nested things mixed in here.

I will revisit this tomorrow and see if I still have problems with it (and if so, hopefully have some suggestions).

Reply:
Getting a better codestyle here (and also below with all the maps there) would be awesome. I'm a rust beginner, so in some places, I basically changed the code until it compiled and worked :-(

Reply:
With a little bit of fiddling, I've been able to get this down to this:

```
       let is_context_match = match_regex(&kube_ctx, &context_config.context_pattern);
        if !is_context_match {
            return false;
        };
        if let Some(user_pattern) = context_config.user_pattern {
            if let Some(user) = kube_user.as_ref() {
                return match_regex(user, user_pattern);
            } else {
                // If there is a user pattern but no user, the pattern cannot match
                return false;
            }
        } else {
            // If there is no user pattern, then this context config matches
            return true;
        }
```

with the following definition of `match_regex`:
```
fn match_regex(target: &str, pattern: &str) -> bool {
    match regex::Regex::new(pattern) {
        Ok(re) => re.is_match(&target),
        Err(error) => {
            log::warn!("Could not compile regex pattern `{}`:\n{}", pattern, error);
            false
        }
    }
}
```

I'm still not entirely happy with this, because:

1. The `match_regex` name is terrible.
2. I think the body of the closure in `find` can still be cleaned up a little bit
3. I think we can modify match_regex to do capture group parsing and output in a single pass instead of having to do two passes

but I think it's a starting point.

Reply:
One downside here is that the error messages won't contain the context of whether it's a context-regex or a user-regex, but I doubt most users will have the same regex for both (and if they do, they both need to be fixed).